zk-token-sdk/src/curve25519/edwards.rs

```diff
@@ -74,8 +75,8 @@ mod target_arch {
 
         fn validate_point(&self) -> bool {
             CompressedEdwardsY::from_slice(&self.0)
-                .decompress()
-                .is_some()
+                .map(|compressed_edwards_y| compressed_edwards_y.decompress().is_some())
+                .unwrap_or(false)
```

Comment:
Hm, I think both syntax have pros and cons. I personally preferred `.ok()/.and_then/.is_some()` since it is more concise, but @ilya-bobyr's point about `let / else` being more readable is also quite true since there is no point of having `Ok(false)` when we can just return `false` earlier in the logic.

Reply:
I find `.ok()/.and_then/.is_some()` chains harder to follow.
Just recently, we had an issue when `ok_or_else()` was used instead of `unwrap_or_else()` and it went unnoticed: https://github.com/anza-xyz/agave/pull/1689

I was suggesting we rewrite it as an `if/else` and even during a discussion of the very same block nobody noticed that the original logic is incorrect: https://github.com/anza-xyz/agave/pull/1301

Reply:
okay. I will try to refactor all logic in this PR to `let / else` style 🫡

Reply:
tried to update all at [23147ed](https://github.com/anza-xyz/agave/pull/1693/commits/23147ed6015ea2ac64b75dc7651b50fca389a05d) but hit some clippy error so do some extra updates [2886fe6](https://github.com/anza-xyz/agave/pull/1693/commits/2886fe63dc3c629867a383f40a27f978c822dc85)